v2/protocol/http/protocol.go

```diff
@@ -221,6 +230,33 @@ func (p *Protocol) Respond(ctx context.Context) (binding.Message, protocol.Respo
 // ServeHTTP implements http.Handler.
 // Blocks until ResponseFn is invoked.
 func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	// Filter the GET style methods:
+	switch req.Method {
+	case http.MethodOptions:
+		if p.OptionsHandlerFn == nil {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		p.OptionsHandlerFn(rw, req)
+		return
+
+	case http.MethodGet:
+		if p.GetHandlerFn == nil {
```

Comment:
This looks wrong: if i have a receiver that doesn't accept events but just creates it, i think get should just work ootb without configuring `GetHandlerFn`

Reply:
what does http.GET do on a push based protocol?

Reply:
Like: https://github.com/cloudevents/sdk-go/pull/498/files#diff-3666ce1687203ab4a81aec71bee67724R81

Reply:
I think it's a reasonable use case to support, given also that is cheap to mantain: eg you want to create an endpoint (like a REST API) that replies with some informations wrapped in a cloudevent

Reply:
oh yeah, added. nice.